File: C++/DSA/Abdul Bari/Recursion/fibonacci.cpp
#include <iostream>
using namespace std;

int fib(int x)
{
    if(x<=1) return x;
    
    return (fib(x-2)+fib(x-1));
}

int main(int sinh, char const *cosech[])
{
    cout<<"6th element of fibonacci series is: "<<fib(6);

    return 0;
}

File: Java/src/main/java/AdvJava/MultiThreading/Fibonacci.java
package AdvJava.MultiThreading;

import java.util.Scanner;

class Fibonacci extends Thread
{
    public void run()
    {
        try
        {
            int a=0, b=1, c=0;
            Scanner sc = new Scanner(System.in);
            System.out.print("Enter the Limit: ");

            int n = sc.nextInt();
            System.out.print("Series: ");
            while (n>0)
            {
                System.out.print(c+" ");
                a=b;
                b=c;
                c=a+b;
                n=n-1;
            }
        }
        catch (Exception ex)
        {
            ex.printStackTrace(System.out);
        }
    }
    public static void main(String[] args)
    {
        System.out.println("Soham Kothari");
        try
        {
            Fibonacci fib = new Fibonacci();
            fib.start();

        }
        catch (Exception ex)
        {
            ex.printStackTrace(System.out);
        }
    }
}
